queries.sql

```diff
@@ -0,0 +1,175 @@
+
+-- SELECT
+
+-- List all organization names.
+SELECT organization.name from organization;
+
+-- OUTPUT
+-- name
+-- -------------
+-- Lambda School
+
+
+-- List all channel names.
+SELECT channel.name from channel;
+
+-- OUTPUT
+-- name
+-- ----------
+-- #general
+-- #random
+
+
+-- List all channels in a specific organization by organization name.
+SELECT channel.name from channel, organization WHERE organization_id = organization.id AND organization.name = "Lambda School";
+
+-- OUTPUT
+-- name
+-- ----------
+-- #general
+-- #random
+
+
+-- List all messages in a specific channel by channel name #general in order of post_time, descending. (Hint: ORDER BY. Because your INSERTs might have all taken place at the exact same time, this might not return meaningful results. But humor us with the ORDER BY anyway.)
+SELECT content FROM message WHERE channel_id IS (SELECT id FROM channel WHERE name IS '#general') ORDER BY post_time DESC;
+
+-- OUTPUT
+-- content
+-- ------------------
+-- Chris is so random
+-- Ha, yeah
+-- I like bread
+
+
+-- List all channels to which user Alice belongs.
+SELECT channel.name, channel.id FROM channel INNER JOIN channel_user ON channel.id IS channel_user.channel_id INNER JOIN user ON user.id IS channel_user.user_id WHERE user.name IS "Alice";
+
+-- OUTPUT
+-- name        id
+-- ----------  ----------
+-- #general    1
+-- #random     2
+
+
+-- List all users that belong to channel #general.
+SELECT user.name from user INNER JOIN channel_user ON user.id IS channel_user.user_id INNER JOIN channel ON channel_user.channel_id IS channel.id WHERE channel.name IS "#general";
+
+-- OUTPUT
+-- name
+-- ----------
+-- Alice
+-- Bob
+
+
+-- List all messages in all channels by user Alice.
+SELECT content from message INNER JOIN user ON message.user_id IS user.id WHERE user.name IS "Alice";
+
+-- OUTPUT 
+-- content
+-- ------------------
+-- Chris is so random
+-- Bob is so basic
+-- What's your favori
+-- ...Can't you take
+
+
+-- List all messages in #random by user Bob.
+SELECT content FROM message WHERE channel_id IS (SELECT id FROM channel WHERE name is "#general") AND user_id IS (SELECT id FROM user WHERE name is "Bob");
+
+-- OUTPUT
+-- content
+-- ----------
+-- Ha, yeah
+-- I like bre
+
+
+-- List the count of messages across all channels per user. (Hint: COUNT, GROUP BY.)
+SELECT user.name AS "User Name", COUNT(user.id) as "Message Count" FROM message INNER JOIN user ON message.user_id IS user.id INNER JOIN channel WHERE message.channel_id IS channel.id GROUP BY user.id ORDER BY user_id DESC;
+
+-- OUTPUT
+-- User Name   Message Count
+-- ----------  -------------
+-- Chris       4
+-- Bob         2
+-- Alice       4
+
+
+-- What SQL keywords or concept would you use if you wanted to automatically delete all messages by a user if that user were deleted from the user table?
+
+PRAGMA foreign_keys = OFF
+DELETE FROM message WHERE message.user_id IS user.id
+
```

Comment:
You can use `ON DELETE CASCADE` to the end of the `message.channel_user` and `channel_user.user_id` line to accomplish targeting a deleted user.